File: src/main/java/me/hhitt/disasters/disaster/impl/HotSun.java
package me.hhitt.disasters.disaster.impl;

import me.hhitt.disasters.arena.Arena;
import me.hhitt.disasters.disaster.Disaster;
import me.hhitt.disasters.storage.file.Configuration;
import me.hhitt.disasters.storage.file.FileManager;
import me.hhitt.disasters.util.Notify;
import org.bukkit.Location;
import org.bukkit.entity.Player;

import java.util.concurrent.CopyOnWriteArrayList;

public class HotSun implements Disaster {

    private final CopyOnWriteArrayList<Player> players = new CopyOnWriteArrayList<>();

    @Override
    public void start(final Arena arena) {
        for (Player player : arena.getPlaying()) {
            players.add(player);
        }
        Notify.disaster(arena, "hot-sun");
    }

    @Override
    public void pulse(final int time) {
        if (time % 2 != 0) {
            return;
        }

        final Configuration cfg = FileManager.get("config");
        final double damageAmount = cfg != null ? cfg.getDouble("hot-sun-damage", 0.5) : 0.5;

        for (Player player : players) {
            final Location loc = player.getLocation();

            final boolean exposed = loc.getBlock().getLightFromSky() >= 15
                && player.getWorld().getHighestBlockAt(loc).getY() <= loc.getBlockY();

            if (exposed) {
                player.damage(damageAmount);
            }
        }
    }

    @Override
    public void stop(final Arena arena) {
        for (Player player : arena.getPlaying()) {
            players.remove(player);
        }
    }
}

File: src/main/kotlin/me/hhitt/disasters/disaster/impl/HotSun.kt
package me.hhitt.disasters.disaster.impl

import me.hhitt.disasters.Disasters
import me.hhitt.disasters.arena.Arena
import me.hhitt.disasters.disaster.Disaster
import me.hhitt.disasters.util.Notify
import org.bukkit.entity.Player
import java.util.concurrent.CopyOnWriteArrayList

class HotSun : Disaster {

    private val players = CopyOnWriteArrayList<Player>()

    // 1. Get the plugin instance so we can access the config
    private val plugin = Disasters.getInstance()

    override fun start(arena: Arena) {
        arena.playing.forEach { players.add(it) }
        Notify.disaster(arena, "hot-sun")
    }

    override fun pulse(time: Int) {
        if (time % 2 != 0) return

        // 2. Read the damage amount from config (defaults to 0.5 if it can't find it)
        val damageAmount = plugin.config.getDouble("hot-sun-damage", 0.5)

        players.forEach { player ->
            val loc = player.location

            val exposed = loc.block.lightFromSky >= 15 &&
                    player.world.getHighestBlockAt(loc).y <= loc.y

            if (exposed) {
                // 3. Apply the configurable damage
                player.damage(damageAmount)
            }
        }
    }


    override fun stop(arena: Arena) {
        arena.playing.forEach { players.remove(it) }
    }
}